Source/JavaScriptCore/API/JSValue.h

```diff
@@ -263,39 +300,49 @@ NS_CLASS_AVAILABLE(10_9, 7_0)
 /*!
 @method
 @abstract Convert a JSValue to a double.
-@discussion The JSValue is converted to a number according to the rules specified 
- by the JavaScript language.
 @result The double result of the conversion.
+@discussion Convert the JSValue to a number according to the rules specified by the JavaScript language. Unless the JSValue is a BigInt then this is equivalent to <code>Number(value)</code> in JavaScript.
```

Comment:
This discussion is ambiguous about what happens if the JSValue is a BigInt.  It doesn't say anything about it.  We should fix that.  According to https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#no-implicit-conversions-or-mixed-operands, the VM would throw a TypeError.  How is that manifested in the API?  Or is that documented elsewhere?

Reply:
Previously, `toDouble` or `toNumber` doesn't support for BigInt. Since we're introducing BigInt API in this patch and we also want to support the conversion from BigInt to double [1], instead of creating a new API like `toBigIntDouble/Number` we decided to extend the existing `toDouble/Number` for BigInt.

[1] https://tc39.es/ecma262/#sec-number-constructor-number-value